src/ruby/puppet-server-lib/puppet/server/http_client.rb

```diff
@@ -70,12 +79,24 @@ def get(url, headers)
     request_options.set_as(ResponseBodyType::TEXT)
     configure_timeouts(request_options)
     configure_ssl(request_options)
-    response = SyncHttpClient.get(request_options)
+    response = client_get(request_options)
     ruby_response(response)
   end
 
   private
 
+  def client_get(request_options)
+    SyncHttpClient.get(request_options)
+  rescue Java::ComPuppetlabsHttpClient::HttpClientException => e
+    raise Puppet::Server::HttpClientError.new(e.inspect, e)
```

Comment:
I went with `inspect` because it's supposed to be the most verbose object->string method, and that's what the classifier code was using to print out this error. Happy to go with something else if we know it's better or more conventional.